Assets/Scripts/UI/Components/ATM/Timeline_Regulator.cs

```diff
@@ -0,0 +1,237 @@
+        private IEnumerator ScrollToMaxYearAfterDelay()
+        {
+            yield return new WaitForEndOfFrame(); // Wait until the end of the frame  
+            ScrollToYear(maxYear); // Now scroll to the maximum year  
+        }
+
+        private IEnumerator RebuildToCurrentAfterDelay()
+        {
+            yield return new WaitForEndOfFrame(); // Wait until the end of the frame  
+            ScrollToYear(currentYear); // Now scroll to the maximum year  
+        }
+
+        void InitializeTimeline()
+        {
+            // Clear previous year instances  
+            foreach (var instance in yearInstances)
+            {
+                Destroy(instance.gameObject);
+            }
+            yearInstances.Clear();
+
+            // Read the years from the input fields  
+            minYear = int.Parse(startYearInput.text);
+            maxYear = int.Parse(endYearInput.text);
+
+            for (int year = minYear; year <= maxYear; year++)
+            {
+                GameObject yearInstance = Instantiate(yearPrefab, content);
+                yearInstance.GetComponentInChildren<TMP_Text>().text = year.ToString();
+                yearInstances.Add(yearInstance.GetComponent<RectTransform>());
+
+                // Handle event years as before  
+                GameObject eventObject = yearInstance.transform.GetChild(1).gameObject; // Assuming event is the second child  
+                if (eventYears.Contains(year))
+                {
+                    eventObject.SetActive(true);
+                    Button eventButton = eventObject.GetComponent<Button>();
+                    if (eventButton != null)
+                    {
+                        int yearToScroll = year; // Store the year in a local variable  
+                        eventButton.onClick.AddListener(() =>
+                        {
+                            ScrollToYear(yearToScroll);
+                        });
+                    }
+                }
+                else
+                {
+                    eventObject.SetActive(false);
+                }
+            }
+
+            ScrollToYear(currentYear);
+        }
+
+        void UpdateTimeline()
+        {
+            // Validate input and initialize timeline with new years  
+            if (int.TryParse(startYearInput.text, out int newMinYear) && int.TryParse(endYearInput.text, out int newMaxYear))
+            {
+                minYear = newMinYear;
+                maxYear = newMaxYear;
+                InitializeTimeline();
+                StartCoroutine(RebuildToCurrentAfterDelay());
+            }
+        }
+
+        void Update()
+        {
+            // Smoothly move to the target position  
+            content.anchoredPosition = Vector2.Lerp(content.anchoredPosition, targetPosition, Time.deltaTime * lerpSpeed);
+
+            // Handle year input submission  
+            if (Input.GetKeyDown(KeyCode.Return))
```

Comment:
Het oude input systeem werkt niet meer in ons project, je moet `Keyboard.current.enterKey.wasPressedThisFrame` gebruiken